### PR TITLE
[MAINTENANCE] Remove suite CRUD from data_context

### PR DESCRIPTION
**Pull request by tyler-hoffman**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/docs/oss/guides/connecting_to_your_data/fluent/database/gcp_deployment_patterns_file_bigquery.py [hunk 1]
```diff
@@ -259,7 +261,10 @@
 
 
 # <snippet name="docs/docusaurus/docs/oss/guides/connecting_to_your_data/fluent/database/gcp_deployment_patterns_file_bigquery.py add_or_update_expectation_suite">
-context.add_or_update_expectation_suite(expectation_suite_name="test_bigquery_suite")
+try:
+    context.suites.add(ExpectationSuite(name="test_bigquery_suite"))
+except DataContextError:
+    ...
```
Comment (cdkini) [inline docs/docusaurus/docs/oss/guides/connecting_to_your_data/fluent/database/gcp_deployment_patterns_file_bigquery.py:267]: Are we okay with this for now or should we have an idomatic pattern for idempotent creation/retrieval?

Comment (tyler-hoffman) [reply docs/docusaurus/docs/oss/guides/connecting_to_your_data/fluent/database/gcp_deployment_patterns_file_bigquery.py:267]: I think we're okay for now. We've said we don't want to continue with the `add_or_update_` pattern any more. I do find the try/accept stuff pretty ugly though.

File: great_expectations/experimental/rule_based_profiler/helpers/util.py [hunk 2]
```diff
@@ -1032,14 +1032,12 @@ def get_or_create_expectation_suite(  # noqa: C901
...
-        if persist:
+        if persist and data_context:
             try:
                 # noinspection PyUnusedLocal
                 expectation_suite = data_context.suites.get(name=expectation_suite_name)
             except gx_exceptions.DataContextError:
-                expectation_suite = data_context.add_expectation_suite(
-                    expectation_suite_name=expectation_suite_name
-                )
+                expectation_suite = data_context.suites.get(expectation_suite_name)
```
Comment (cdkini) [inline great_expectations/experimental/rule_based_profiler/helpers/util.py]: is this correct? we do the same logic in the except block?

**Review by cdkini — APPROVED** [review]

File: great_expectations/experimental/rule_based_profiler/helpers/util.py (hunk 2 above)
Comment (tyler-hoffman) [reply great_expectations/experimental/rule_based_profiler/helpers/util.py]: great catch! switched it to an add.

**Pull request merged**